### DISCUSSION TITLE
How to persist jobs data outside the docker container

### DISCUSSION THREAD

**Discussion by AhmedAbouelkher** [dkron-io/dkron#1277] [Q&A] [answered]
I read the docs about [Mounting a mapped file storage volume](https://dkron.io/docs/basics/installation#mounting-a-mapped-file-storage-volume) and mounted an external volume to save my data.

For some reason, the only data saved are for the raft protocol, So if for example, I needed to move my data to another server, all my jobs are gone because the container itself was destroyed.

Is there a better way to fix this? or backup the jobs?

**Comment by yvanoers** [comment, marked as answer]
To answer the backup question: https://dkron.io/docs/usage/upgrade/#backup--restore
The same page explains a online upgrade process that would work just as well for moving dkron to another server if the dkron instances can talk to each other.

The raft log is all that is persisted and should be all you need. Dkron rebuilds its state from the log at startup.
If moving the data folder didn't restore the state, I think that might be because it 'knowns' it is not running as the same node. But I'm not 100% on the raft workings off the top of my head.

**Reply by AhmedAbouelkher** [reply]
Backing up jobs data manually is not practical in my case.
For some reason when running `docker compose down` all the data on the container are gone even when I [mounted the storage volume](https://dkron.io/docs/basics/installation#mounting-a-mapped-file-storage-volume).

**Reply by vcastellm** [reply]
You can not move the Raft data folder to another node, @yvanoers is correct, you need to backup your jobs and restore them in the new node, otherwise you should rely on another source for the jobs and post them to Dkron using the API.
